src/main/kotlin/org/rust/ide/surroundWith/expression/RustExpressionSurrounderBase.kt

```diff
@@ -20,7 +24,15 @@ abstract class RustExpressionSurrounderBase : Surrounder {
     }
 
     final override fun surroundElements(project: Project, editor: Editor, elements: Array<out PsiElement>): TextRange? {
-        return surroundExpression(project, editor, requireNotNull(targetExpr(elements)))
+        val expression = requireNotNull(targetExpr(elements))
+        val templateExpr = RustElementFactory.createExpression(project, getTemplate()) as RustExprElement
+
+        getLeafExpression(templateExpr).replace(expression)
+        val newExpression = expression.replace(templateExpr)
+
+        CodeInsightUtilCore.forcePsiPostprocessAndRestoreElement(newExpression)
+
+        return getSelectionRange(newExpression)
```

Comment:
>Also I can keep the existing RustExpressionSurrounderBase, and derive RustSimpleExprSurrounderBase.

Will there be any classes which inherent from `RustExpressionSurrounderBase`, but not from `RustSimpleExprSurrounderBase`? If the answer is no, it's better not to do this right now. 

>I wanted to use the same strategy for statements surrounder. Is it good way?

Yep, it looks reasonable!